### PR TITLE
Enable nullability in some Control members, ControlTabOrderHolder and ControlTabOrderComparer

### PR DESCRIPTION
**Pull request by gpetrou**
## Proposed changes

- Enable nullability in some Control members, ControlTabOrderHolder and ControlTabOrderComparer.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/7462)

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/Control.ControlTabOrderComparer.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    public partial class Control
+    {
+        /// <summary>
+        ///  Used to sort controls based on tab index and z-order.
+        /// </summary>
+        private class ControlTabOrderComparer : IComparer<ControlTabOrderHolder>
+        {
+            public int Compare(ControlTabOrderHolder? x, ControlTabOrderHolder? y)
+            {
+                if (x is null && y is null)
```
[the remark this answers is not in the document]
Comment (gpetrou) [reply src/System.Windows.Forms/src/System/Windows/Forms/Control.ControlTabOrderComparer.cs:16]: The interface requires x and y to be nullable, but the current code does not account for that. So, I used the same code we have in AttributeTypeSorter class.

File: src/System.Windows.Forms/src/System/Windows/Forms/Control.cs [hunk 2]
```diff
@@ -6595,8 +6594,13 @@ private static void InvokeMarshaledCallbackHelper(object obj)
             }
         }
 
-        private static void InvokeMarshaledCallbackDo(ThreadMethodEntry tme)
+        private static void InvokeMarshaledCallbackDo(ThreadMethodEntry? tme)
```
[the remark this answers is not in the document]
Comment (gpetrou) [reply src/System.Windows.Forms/src/System/Windows/Forms/Control.cs:6597]: If we revert this change, there is a warning a few lines above in `InvokeMarshaledCallbackDo(tme);`.

File: src/System.Windows.Forms/src/System/Windows/Forms/Control.cs [hunk 3]
```diff
@@ -6690,15 +6697,18 @@ private void InvokeMarshaledCallbacks()
...
-                lock (_threadCallbackList)
+                if (_threadCallbackList is not null)
                 {
-                    if (_threadCallbackList.Count > 0)
-                    {
-                        current = _threadCallbackList.Dequeue();
-                    }
-                    else
+                    lock (_threadCallbackList)
                     {
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/Control.cs:6703]: Presumably we can only get here after `MarshaledInvoke` was called which initialises `_threadCallbackList`, and we never assign `_threadCallbackList` to null. So presumably there's no need for the 2nd null-check inside the lock.

**Review by RussKie — COMMENTED** [review]
LGTM

File: src/System.Windows.Forms/src/System/Windows/Forms/Control.cs (hunk 3 above)
Comment (gpetrou) [reply src/System.Windows.Forms/src/System/Windows/Forms/Control.cs:6703]: Without this check there is still a warning. Do you mean you want to use the null forgiving operator instead?

File: src/System.Windows.Forms/src/System/Windows/Forms/Control.ControlTabOrderComparer.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (dreddy-work) [reply src/System.Windows.Forms/src/System/Windows/Forms/Control.ControlTabOrderComparer.cs:16]: Okay.

**Review by dreddy-work — APPROVED** [review]

**Pull request merged**